src/GafferScene/ShufflePrimitiveVariables.cpp

```diff
@@ -0,0 +1,133 @@
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "GafferScene/ShufflePrimitiveVariables.h"
+
+#include "GafferScene/SceneAlgo.h"
+
+#include "IECoreScene/Primitive.h"
+
+using namespace std;
+using namespace IECore;
+using namespace IECoreScene;
+using namespace Gaffer;
+using namespace GafferScene;
+
+GAFFER_GRAPHCOMPONENT_DEFINE_TYPE( ShufflePrimitiveVariables );
+
+size_t ShufflePrimitiveVariables::g_firstPlugIndex = 0;
+
+ShufflePrimitiveVariables::ShufflePrimitiveVariables( const std::string &name ) : Deformer( name )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+	addChild( new ShufflesPlug( "shuffles" ) );
+}
+
+ShufflePrimitiveVariables::~ShufflePrimitiveVariables()
+{
+}
+
+Gaffer::ShufflesPlug *ShufflePrimitiveVariables::shufflesPlug()
+{
+	return getChild<Gaffer::ShufflesPlug>( g_firstPlugIndex );
+}
+
+const Gaffer::ShufflesPlug *ShufflePrimitiveVariables::shufflesPlug() const
+{
+	return getChild<Gaffer::ShufflesPlug>( g_firstPlugIndex );
+}
+
+bool ShufflePrimitiveVariables::affectsProcessedObject( const Gaffer::Plug *input ) const
+{
+	return Deformer::affectsProcessedObject( input ) || shufflesPlug()->isAncestorOf( input );
+}
+
+void ShufflePrimitiveVariables::hashProcessedObject( const ScenePath &path, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	Deformer::hashProcessedObject( path, context, h );
+
+	shufflesPlug()->hash( h );
+}
+
+IECore::ConstObjectPtr ShufflePrimitiveVariables::computeProcessedObject( const ScenePath &path, const Gaffer::Context *context, const IECore::Object *inputObject ) const
+{
+	auto inputPrimitive = runTimeCast<const Primitive>( inputObject );
+	if( !inputPrimitive )
+	{
+		return inputObject;
+	}
+
+	if( shufflesPlug()->children().empty() )
+	{
+		return inputObject;
+	}
+
+	PrimitivePtr result = inputPrimitive->copy();
+	shufflesPlug()->shuffle<PrimitiveVariableMap>( inputPrimitive->variables, result->variables );
+
+	return result;
+}
+
+bool ShufflePrimitiveVariables::adjustBounds() const
+{
+	if( !Deformer::adjustBounds() )
+	{
+		return false;
+	}
+
+	for( auto &child : shufflesPlug()->children() )
+	{
+		ShufflePlug *shuffle = static_cast<ShufflePlug *>( child.get() );
+
+		if( shuffle->enabledPlug()->getValue() )
+		{
+			std::string destination = shuffle->destinationPlug()->getValue();
+			if( destination == "P" )
+			{
+				return true;
+			}
+			else if( shuffle->sourcePlug()->getValue() == "P" )
+			{
+				if( shuffle->deleteSourcePlug()->getValue() || destination == "${source}" || destination == "$source" )
+				{
+					return true;
+				}
+			}
```

Comment:
I wouldn't worry about this because deleting P makes little sense anyway. For primitives that need P, they'd become invalid (and therefore guaranteed not to extend outside the old bound), and for primitives that don't need P, it would have no effect on the bound.

Reply:
Addressed in e810578